backend/data-store/db-client/mail.go

```diff
@@ -0,0 +1,265 @@
+package dbclient
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+	ds "zpi/pb"
+	queries "zpi/sql/gen"
+
+	"github.com/samber/lo"
+	"go.uber.org/zap"
+)
+
+type (
+	EmailRow struct {
+		AssessorID     int32
+		SubmissionID   int32
+		SubmissionName string
+		Year           int32
+		SetBeforeDate  sql.NullTime
+		IsCreated      bool
+		FirstName      string
+		LastName       string
+		Email          string
+
+		RatingType    ds.RatingType
+		RatingSqlType queries.ProjectRatingType
+	}
+)
+
+func (st *Store) GetEmailDetails(ct context.Context) (*ds.EmailResponse, error) {
+	var (
+		ctx, cancel       = context.WithTimeout(ct, 30*time.Minute)
+		wg                = sync.WaitGroup{}
+		individualRatings = []*EmailRow{}
+		finalRatings      = []*EmailRow{}
+		initialRatings    = []*EmailRow{}
+	)
+	defer cancel()
+	wg.Add(3)
+
+	go func() {
+		defer wg.Done()
+		emails, err := getWithBackoff(ctx, st.Log, 3, st.getEmailsForFinalRatings)
+		if err != nil {
+			st.Log.Error("error getting emails", zap.Error(err))
+			return
+		}
+		finalRatings = *emails
+	}()
+
+	go func() {
+		defer wg.Done()
+		emails, err := getWithBackoff(ctx, st.Log, 3, st.getEmailsForIndividualRatings)
+		if err != nil {
+			st.Log.Error("error getting emails", zap.Error(err))
+			return
+		}
+		individualRatings = *emails
+	}()
+
+	go func() {
+		defer wg.Done()
+		emails, err := getWithBackoff(ctx, st.Log, 3, st.getEmailsForInitialRatings)
+		if err != nil {
+			st.Log.Error("error getting emails", zap.Error(err))
+			return
+		}
+		individualRatings = *emails
+	}()
+
+	wg.Wait()
+
+	merged := append(append(individualRatings, finalRatings...), initialRatings...)
+	emailsToSend := make([]*ds.EmailDetails, 0, len(merged))
+	dbConn := queries.New(st.Pool)
+
+	for _, email := range merged {
+
+		if !email.SetBeforeDate.Valid { // Unreachable if I'm sane
+			st.Log.Error("email has no set before date", zap.Any("email", email))
+			continue
+		}
+
+		var ( // Are we ...
+
+			pastWarnings = time.Now().After(email.SetBeforeDate.Time)
+
+			// Let's say:
+			// today               = 13.05
+			// deadline date       = 15.05
+			// then:
+			// warningPeriodStart = deadline - EmailWarningPeriod (5 days) = 10.05
+			// is warningPeriodStart (10.05) before now (13.05)?
+			//    - yes, so we are in EmailWarningPeriod
+			//    - no,  so we are not in EmailWarningPeriod
+			inWarningPeriod      = email.SetBeforeDate.Time.Add(-EmailWarningPeriod).Before(time.Now())
+			inFinalWarningPeriod = email.SetBeforeDate.Time.Add(-EmailWarningFinalPeriod).Before(time.Now())
```

Comment:
Any reason not to use this - https://pkg.go.dev/time#Time.Sub

Reply:
This is for subtracting operation `date - date`. We need `date - duration`. 
Take a look at the last sentence of what you linked 
> To compute t-d for a duration d, use t.Add(-d).

Reply:
Thanks for clarifying, without the explicit types I'm dumb.